docs/derivation.md

```diff
@@ -0,0 +1,106 @@
+
+It can be very challenging to reconciliate the real world mistakes back into a well specified derivation. We can solve it with our current approach to an extend, if a user is stuck in a specific path and want to use Ledger Live, he will have to move out the funds to a better path (using the original tool he used to derives address in the first place).
+
+**When we integrate a new coin, we will always to try to converge the existing approaches to the most common used paradigm and will always prefer BIP 44 when possible.** Sometimes we support more than one, for instance when we integrated Tezos there where both tezbox and galleon that had different _default_ derivation, we found a way to scan for both.
+
+All exceptions are written in [src/derivation.js](../src/derivation.js).
+
+## DerivationMode ?
+
+derivation.js exposes a type called `DerivationMode`. It's essentially an enum string of all possible derivation schemes.
+
+By convention we use `""` to describe the legacy 44 purpose. (it's essentially the "default"). Otherwise we have `"segwit"` and `"native_segwit"`, but many more.
+
+The DerivationMode is used for many things and is part of the Account information. (it's contained in the account `id`)
+
+Most of the functions that are exposed by derivation.js will take a DerivationMode in parameter.
+
+## getDerivationModesForCurrency
+
+The starting point of derivation.js is essentially `getDerivationModesForCurrency`. This function is used by the `bridge.scanAccounts` logic to know where to derive accounts. It gives for a given crypto currency the array of "DerivationMode" to scan for.
+
+So for instance, for Bitcoin we described there were 3 classical cases, that function would just return `[ "", "segwit", "native_segwit" ]`.
+
+We have however also the concept of "legacy" derivations. In case of Bitcoin, we also will prepend a `"legacy_on_bch"` which are to recover a common case of people sending Bitcoin to Bitcoin Cash addresses.
+
+We have also some special "custom" accounts (by enabling _Experimental Settings > Extended account search_) that would enable more derivation modes:
+
+```
+[ "legacy_on_bch", "segwit_on_legacy", "legacy_on_segwit", "", "segwit", "native_segwit" ]
+```
+
+`segwit_on_legacy` and `legacy_on_segwit` are ways to recover from classical user mistake of having sent funds on a Segwit addressed derived on a legacy account, or reversely. That way allows these users to create (in advanced mode way) the accounts to recover the funds.
+
+The returned value really different for each coin, because some coins supports segwit some don't, same for native segwit. Some coins also are **hard fork of Bitcoin** which force us to support a special derivation (we called `"unsplit"`, because it's the derivation path before splitting the funds out of the parent derivation path).
```

Comment:
either "The returned values *are* really different for each coin" or "The returned values really differ for each coin"